### PR TITLE
doc: update release schedule

### PR DESCRIPTION
**Pull request by jychen7**
# Which issue does this PR close?

part of https://github.com/apache/arrow-datafusion/issues/2095

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: dev/release/README.md [hunk 1]
```diff
@@ -32,9 +32,17 @@ Python binding or Ballista always requires a new DataFusion version release.
 
 ## Branching
 
-DataFusion currently only releases from the `master` branch. Given the project
-is still in early development state, we are not maintaining an active stable
-release backport branch.
+### Major Release
+
+DataFusion have major releases from the `master` branch every 3 months.
```
Comment (alamb) [inline dev/release/README.md:37]:
```suggestion
DataFusion typically has major releases from the `master` branch every 3 months, including breaking API changes. 
```

File: dev/release/README.md [hunk 2]
```diff
@@ -32,9 +32,17 @@ Python binding or Ballista always requires a new DataFusion version release.
...
+
+### Minor Release
+
+Starting v7.0.0, we experiments on maintaining an active stable release branch (e.g. `maint-7.x`). Every month, we reviews the `maint-*` branch and prepares a minor release (e.g. v7.1.0) when necessary. A patch release (v7.0.1) can be requested on demand if it is urgent bug/security fix.
```
Comment (alamb) [inline dev/release/README.md:41]:
```suggestion
Starting v7.0.0, we are experimenting with maintaining an active stable release branch (e.g. `maint-7.x`). Every month, we will review the `maint-*` branch and prepare a minor release (e.g. v7.1.0) when necessary. A patch release (v7.0.1) can be requested on demand if it is urgent bug/security fix.
```

File: dev/release/README.md [hunk 3]
```diff
@@ -32,9 +32,17 @@ Python binding or Ballista always requires a new DataFusion version release.
...
+
+How to add changes to `maint-*` branch?
+1. follow normal workflow to create PR to `master` branch and wait for its approval and merges.
+2. after PR is sqush merged to `master`, branch from most recent maintaince branch (e.g. `maint-7-x`), cherry-pick the commit and create a PR to maintaince branch (e.g. `maint-7-x`).
```
Comment (alamb) [inline dev/release/README.md:45]:
```suggestion
#### How to add changes to `maint-*` branch?
If you would like to propose your change for inclusion in the maintenance branch
1. follow normal workflow to create PR to `master` branch and wait for its approval and merges.
2. after PR is squash merged to `master`, branch from most recent maintenance branch (e.g. `maint-7-x`), cherry-pick the commit and create a PR to maintenance branch (e.g. `maint-7-x`).
```

**Review by alamb — APPROVED** [review]

**Review by yjshen — APPROVED** [review]

**Comment by yjshen** [comment]
Thanks @jychen7 for the doc and volunteering for further releases!

**Pull request merged**